src/object/srv_obj.c

```diff
@@ -407,9 +407,12 @@ ds_obj_update_sizes_in_reply(crt_rpc_t *rpc)
 
 /**
  * Pack nrs in sgls inside the reply, so the client can update
- * sgls before it returns to application. Note: this is only
- * needed for bulk transfer, for inline transfer, it will pack
- * the complete sgls inside the req/reply, see obj_shard_rw().
+ * sgls before it returns to application.
+ * Pack sgl's data size in the reply, client fetch can based on
+ * it to update sgl's iov_len.
+ *
+ * Note: this is only needed for bulk transfer, for inline transfer,
+ * it will pack the complete sgls inside the req/reply, see obj_shard_rw().
```

Comment:
Why don't we do the same thing to bulk? That'll be more straightforward and flexible, also less code (inline and bulk can share same code, right?

Reply:
see above comment, but not sure it answered your question